simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -804,13 +812,26 @@ class simd<std::uint64_t, simd_abi::avx2_fixed_size<4>> {
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd& operator=(simd const&) = default;
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd& operator=(simd&&) = default;
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION static constexpr std::size_t size() {
-    return 8;
+    return 4;
   }
   template <class U, std::enable_if_t<std::is_convertible_v<U, value_type>,
                                       bool> = false>
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd(U&& value)
       : m_value(_mm256_set1_epi64x(bit_cast<std::int64_t>(value_type(value)))) {
   }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd(std::uint64_t a, std::uint64_t b,
+                                             std::uint64_t c, std::uint64_t d)
+      : m_value(_mm256_setr_epi64x(a, b, c, d)) {}
```

Comment:
This constructor is not mentioned in the API and does not exist in the TS Extensions for parallelism v2.
I don't think it is OK to add.

Reply:
Agreed, I think @crtrott and I talked about not having this type of constructor on any of the SIMD types (I added it to some but not others)

Reply:
This constructor is specifically added just to enable multiplication op testing for uint64_t in AVX2 backend. This is because an instruction for 64bit ints multiplication does not exist in AVX2 instruction set and that a fallback approach is currently in place for int64_t.
https://github.com/kokkos/kokkos/blob/9480cb5f8ce56f545f6b693993dad31056d00a4f/simd/src/Kokkos_SIMD_Common.hpp#L129-L137
But I also agree that this does not seem to belong in the public API.

Reply:
these are not the same constructor. @dalg24 is pointing to a constructor that takes four `uint64_t` values. You are referring to the constructor that takes a generator (functor).

Reply:
Removed this type of constructor from all simd types in [2d13485](https://github.com/kokkos/kokkos/pull/5913/commits/2d134858504e04b50eebb2a53b3799c0f523d7a0).